src/Components/WebAssembly/Server/src/ComponentsWebAssemblyEndpointRouteBuilderExtensions.cs

```diff
@@ -107,5 +128,71 @@ private static void AddMapping(FileExtensionContentTypeProvider provider, string
                 provider.Mappings.Add(name, mimeType);
             }
         }
+
+        private static void NegotiateEncoding(HttpContext context, IWebHostEnvironment webHost)
+        {
+            var accept = context.Request.Headers[HeaderNames.AcceptEncoding];
+            if (StringValues.IsNullOrEmpty(accept))
+            {
+                return;
+            }
+
+            if (!StringWithQualityHeaderValue.TryParseList(accept, out var encodings) || encodings.Count == 0)
+            {
+                return;
+            }
+
+            var supportedEncodings = new HashSet<StringSegment>(StringSegmentComparer.OrdinalIgnoreCase) { "gzip" };
```

Comment:
This `HashSet<T>` is never mutated, so is it worth turning it into a shared `static` instance to avoid reallocating/repopulating for each request?

Reply:
I've moved this into a static.